consensus/clique/clique.go

```diff
@@ -324,6 +324,12 @@ func (c *Clique) verifyCascadingFields(chain consensus.ChainHeaderReader, header
 	if parent.Time+c.config.Period > header.Time {
 		return errInvalidTimestamp
 	}
+	// Verify the header's EIP-1559 attributes.
+	if chain.Config().IsAleut(header.Number) {
```

Comment:
```suggestion
	if chain.Config().Is1559(header.Number) {
```
Maybe we can make things simpler for ourselves by doing this: 

- The `evm.chainRules` has booleans for `isEip1559` etc. 
- The `param/config: ChainConfig` has `IsEip1559()`. Which we implement like this: 

```golang
// Is1559 returns whether 1558 is activated
func (c *ChainConfig) Is1559(num *big.Int) bool {
		return c.IsAleut(num) || c.IsLondon(num)
}

// IsAleut returns whether num represents a block number after the Aleut fork
func (c *ChainConfig) IsAleut(num *big.Int) bool {
	return isForked(c.AleutBlock, num)
}
```
Since we may have several things going into an upgrade, it would be nice if each EIP implementation would just add whatever it needs, and then some other "framework" PR could add the logic mapping eips to hardforks. 

The way it is written (and also how it's been done several times before), it becomes a bit painful to replace `Aleut` with something else later on, and makes it cumbersome if we want to have e.g. both aleut and london defined.

Reply:
Good idea.